### DISCUSSION TITLE
Specials: maintain special price as percentage when defined as a percentage.

### DISCUSSION THREAD

**Discussion by torvista** [zencart/zencart#5803] [Ideas]
When a Special is defined as a percentage, the special price is calculated on the price at that moment and the special price is stored as a price, not as a percentage. 
Subsequently if the product price is changed (by a csv import for example), the special price does not change and so the percentage is not maintained.
If the Special definition offers the option of absolute price or percentage, the percentage should always be a percentage.

**Comment by torvista** [comment]
I added an extra column that stores the original modifier used (absolute new price or percentage) and added a button on the main Specials page to recalculate all Specials prices that were based on a modifier that was a percentage.

Aside, perhaps there is also a case to allow a modifier that is a fixed discount off the price (like -10), that could also be maintained through price updates.

**Comment by scottcwilson** [comment]
This is a big change that others might disagree with.  Don't want to pursue.

**Comment by dbltoe** [comment]
It's always been a thorn in our side as most sites we've built or host use percent and have to jump through hoops with each MSRP change.

**Comment by mc12345678** [comment]
> This is a big change that others might disagree with. Don't want to pursue.

The decision on this appears absolutely and unnecessarily rushed.

If it's not already there, add another on-screen selector to make/allow the percentage to be a constant percentage versus a one time quick calculation. I've always thought it odd not to be part of the base code. 

That someone stepped up to make the change and offer it plus the support and be handed the above response?  I feel bad for the recipient, hope they consider continuing to develop for Zen Cart.

**Comment by torvista** [comment]
![image](https://github.com/zencart/zencart/assets/4391026/d734f165-7edf-4e99-854e-14cf34f371b2)

In the flurry of spring-cleaning going on now (for some reason), this is throwing the baby out with the bath water.

**Comment by scottcwilson** [comment]
Please change your store as you wish. 
If you think this is a great idea, build a plugin. 
If the plugin gets a ton of adoption, we can look at pulling it into the core. 

OR

When you bulk upload price changes from a vendor, update special prices at the same time

The reason for cleaning up the issues list is to keep it focused on *actual issues* that we have some plan to address at some point.  It's not intended to be a dumping ground of "I wish it worked this way" posts.

**Comment by torvista** [comment]
This can also be moved to Discussions.

**Comment by torvista** [comment]
Done:
https://github.com/torvista/Zen_Cart-Specials_Modifier
